L1Trigger/L1TGlobal/plugins/GenToInputProducer.cc

```diff
@@ -259,6 +271,23 @@ namespace l1t {
       LogTrace("GtGenToInputProducer") << ">>> GenParticles collection not found!" << std::endl;
     }
 
+    // Muon Shower Collection
+
+    bool mus0 = (bool)mushowerRandom->Integer(2);  // should be [0,1] = 1 bit;
+    bool mus1 = (bool)mushowerRandom->Integer(2);
+    bool mus2 = (bool)mushowerRandom->Integer(2);
+    bool mus2Loose = (bool)mushowerRandom->Integer(2);
+    bool musoot0 = (bool)mushowerRandom->Integer(2);
+    bool musoot1 = (bool)mushowerRandom->Integer(2);
+    bool musoot2Loose = (bool)mushowerRandom->Integer(2);
```

Comment:
Am I misunderstanding something about why these seem to be set with just random stuff? Is this just a test of whatever random inputs we try to put in?

Reply:
Hi @aloeliger, yes, the idea of GenToInputProducer is actually to give the possibility to generate random ("fake") inputs, and it is in fact use within the script https://github.com/cms-sw/cmssw/blob/master/L1Trigger/L1TGlobal/test/runGlobalFakeInputProducer.py.